File: KrasiMilan/QL/src/main/java/nl/uva/softwcons/ql/ast/expression/Expression.java
package nl.uva.softwcons.ql.ast.expression;

import nl.uva.softwcons.ql.ast.ASTNode;
import nl.uva.softwcons.ql.ast.LineInfo;

public abstract class Expression implements ASTNode {
    private final LineInfo lineInfo;

    public Expression(final LineInfo lineInfo) {
        this.lineInfo = lineInfo;
    }

    public abstract <T> T accept(ExpressionVisitor<T> visitor);

    public LineInfo getLineInfo() {
        return lineInfo;
    }
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expression.cs
using AST.Nodes.Interfaces;
using AST.Representation;

namespace AST.Nodes
{
    public abstract class Expression : ASTNode, IVisitable
    {
        public Expression(PositionInText position)
            : base(position) { }

        public abstract T Accept<T>(ASTVisitors.IVisitor<T> visitor);
    }
}
